### PR TITLE
Add a nodepool check to datadog

### PR DESCRIPTION
**Pull request by loafyloaf**
This will add a datadog check for the nodepool status page.

This time with license info!

### CODE REVIEW

**Review by omgjlk — APPROVED** [review]

**Comment by omgjlk** [comment]
recheck

**Review by jamielennox — CHANGES_REQUESTED** [review]
So i'm interested why the two reports are events and not gauge or status_check.

To me the most important thing that nodepool can report here is a self.service_check(). The service check is what says nodepool is up and responding, and then events and stats are reported within that.

So IMO a timeout_event here, or a status_code != 200 event is a negative service_check rather than an event.

The information I expected to get back from the status URL then is like, how many images are currently available or build errors - although looking at the response to image-list that information may not be available. 

I don't know if the response time to a local url is really a useful stat.

File: roles/dd-nodepool/files/nodepool_server.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        self.event({
+            'timestamp': int(time.time()),
+            'event_type': 'http_check',
+            'msg_title': 'URL timeout',
+            'msg_text': '%s timed out after %s seconds.' % (status_url, timeout),
+            'aggregation_key': aggregation_key
+        })
+
+    def status_code_event(self, status_url, r, aggregation_key):
+        self.event({
```
Comment (jamielennox) [inline roles/dd-nodepool/files/nodepool_server.py]: So i'm interested and not sure why these 2 are events

File: roles/dd-nodepool/files/nodepool_server.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        if 'status_url' not in instance:
+            self.log.info("Skipping instance, no url found.")
+            return
+
+        status_url = instance['status_url']
+        default_timeout = self.init_config.get('default_timeout', 5)
+        timeout = float(instance.get('timeout', default_timeout))
+        start_time = time.time()
+        try:
+            r = requests.get(ststus_url, timeout=timeout)
```
Comment (jamielennox) [inline roles/dd-nodepool/files/nodepool_server.py]: status_url

**Review by jamielennox — DISMISSED** [review]

**Pull request merged**